src/tally.rs

```diff
@@ -33,61 +58,73 @@ async fn days() -> Result<(Vec<(PathBuf, u32)>, Vec<u32>), AocError>
         }
     }
 
-
     Ok((have, dont_have))
 }
 
-async fn build_day(day: u32) -> Result<(), AocError>
+async fn build_days(cargo_folder: PathBuf, days: &[(PathBuf, u32)]) -> Result<Vec<u32>, AocError>
 {
-    let bin = format!("day_{:02}", day);
-    let res = tokio::process::Command::new("cargo")
-        .args(["build", "--release", "--bin", &bin])
-        .output()
-        .await?;
+    let sty = ProgressStyle::with_template(
+        "[{elapsed_precise}] {msg}... {bar:40.cyan/blue} {pos:>7}/{len:7}",
+    )
+    .unwrap()
+    .progress_chars("##-");
 
-    if !res.status.success()
-    {
-        Err(AocError::BuildError(bin.to_string()))
-    }
-    else
-    {
-        Ok(())
-    }
-}
+    let progress = ProgressBar::new(days.len() as u64);
+    progress.set_style(sty);
+    progress.set_message("compiling");
 
-async fn build_days(cargo_folder: PathBuf, days: &[(PathBuf, u32)]) -> Result<Vec<u32>, AocError>
-{
-    let mut set = JoinSet::new();
-    for (_path, day) in days.iter()
-    {
+    let res: Result<(), AocError> = thread_exec(days, |(_path, day)| {
         let day = *day;
```

Comment:
not needed as `&u32` also implements display

Reply:
What do you mean?

Reply:
We don't need the extra `let day = *day` since we only use `day` inside `format!()`. So we don't need the shadowing where we dereference `day`. So we can just use day coming from the closure argument and remove the line `let day = *day`

Reply:
I see, I got confused because I though you referred to 
```rs
let res: Result<(), AocError> = thread_exec(days, |(_path, day)| {
```